o1vm/src/mips/constraints.rs

```diff
@@ -511,28 +517,30 @@ impl<Fp: Field> InterpreterEnv for Env<Fp> {
                 ],
             ));
         }
-        // COMMUNICATION CHANNEL: Read hash output
+        // COMMUNICATION CHANNEL: Read hash output FIXME: is it a problem that
```

Comment:
Nit: I suppose the `FIXME` was supposed to be on the next line?

Reply:
To answer to the FIXME: we do only care about the 248 first bits. The last byte is reserved by OP.

Reply:
Yes, but the interpreter is storing the hash in 8 chunks of 4 bytes each. I'll work on a test with your framework to understand if the leading chunk includes the whole 4 bytes or just 3, leaving the 32nd byte empty (for the 0x02). If that was the case, then there is no need to merge https://github.com/o1-labs/proof-systems/pull/2278 (only valid if the real 32 digest is stored). If that is not the case, then we should think of a different way to store the hash bytes on the MIPS side so we can access bytes independently and leave the 32nd outside of the channel (or have extra columns to constrain the decomposition of the most significant chunk into 1 and 3 bytes).